client/src/Components/Pans.js

```diff
@@ -1,15 +1,45 @@
-import React, { useState } from "react";
+import React, { useState, useContext } from "react";
+import { GlobalContext } from "../Contexts/GlobalContext";
 import EditImg from "../Assets/img/icons8-edit(1).svg";
 import ScheduleImg from "../Assets/img/schedule.svg";
+import ArchiveImg from "../Assets/img/archive.png";
+import RestoreImg from "../Assets/img/restore.png";
 import EditPopUp from "./EditTask/EditPopUp";
+
 const Pans = (props) => {
   const [openEditPan, setOpenEditPan] = useState(false);
   const [taskSelected, setTaskSelected] = useState([]);
-  const editPupHandle = (e, task) => {
+  const { setTasksData } = useContext(GlobalContext);
+
+  const handleEditPopup = (e, task) => {
     e.stopPropagation();
     setTaskSelected(task);
     setOpenEditPan(true);
   };
+
+  const handleArchiveTask = (e, task) => {
+    e.stopPropagation();
+
+    const taskArchived = task.task_archived ? false : true;
+
+    fetch("/api/tasks/archived", {
+      method: "PUT",
+      body: JSON.stringify({
+        task_id: task.id,
+        task_archived: taskArchived,
+      }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    fetch("/api/tasks")
```

Comment:
Might want to look into using async await. It's a more modern way of using javascript. Although there is nothing wrong with using .then(). You'll benefit from also knowing async await.